gcp_airflow_foundations/base_class/source_template_config.py

```diff
@@ -0,0 +1,152 @@
+        hds_config : HDS table configuration. See :class:`gcp_airflow_foundations.base_class.hds_table_config.HdsTableConfig`.
+        template_ingestion_options: Configuration for template-level ingestion.
+        facebook_table_config: Extra options for ingesting data from the Facebook API.
+        extra_options: Field for storing additional configuration options.
+        start_date : Start date override for DAG
+        start_date_tz : Timezone
+        version : The Dag version for the table. Can be incremented if logic changes.
+        catchup : Passed to a dag [see doc](https://airflow.apache.org/docs/apache-airflow/stable/dag-run.html#catchup).
+            Defaults to True. May want to change it to False if Dag version is changed, and we don't want to rerun past dags.
+    """
+
+    ingestion_type: IngestionType  # FULL or INCREMENTAL
+    dest_table_override: Optional[str]
+    surrogate_keys: List[str]
+    column_mapping: Optional[dict]
+    cluster_fields: Optional[List[str]]
+    landing_zone_table_name_override_template: Optional[str]
+    column_casting: Optional[dict]
+    new_column_udfs: Optional[dict]
+    hds_config: Optional[HdsTableConfig]
+    facebook_table_config: Optional[FacebookTableConfig]
+    start_date: Optional[str]
+    extra_options: dict = field(default_factory=dict)
+    table_names: list = field(default_factory=list)
+    start_date_tz: Optional[str] = "EST"
+    ods_config: Optional[OdsTableConfig] = OdsTableConfig(
+        ods_metadata=OdsTableMetadataConfig(),
+        ods_table_time_partitioning=None,
+        partition_column_name=None,
+    )
+    template_ingestion_options: TemplateIngestionOptionsConfig = TemplateIngestionOptionsConfig(
+        ingest_all_tables=False,
+        ingestion_name="",
+        dag_creation_mode="TABLE",
+        regex_pattern=""
+    )
+    version: int = 1
+    catchup: bool = True
+
+    # Override values
+    def __post_init__(self):
+        if self.extra_options is None:
+            self.extra_options = {}
+
+        if self.landing_zone_table_name_override_template is None:
+            self.landing_zone_table_name_override_template = "{table}"
+
+    @validator("start_date")
+    def valid_start_date(cls, v):
+        if v is not None:
+            assert datetime.datetime.strptime(
+                v, "%Y-%m-%d"
+            ), "The date format for Start Date should be YYYY-MM-DD"
+        return v
+
+    @root_validator(pre=True)
+    def valid_hds_dataset(cls, values):
+        if (values["cluster_fields"] is not None) and (
+            values["column_mapping"] is not None
+        ):
+            values["cluster_fields"] = [
+                values["column_mapping"].get(field, field)
+                for field in values["cluster_fields"]
+            ]
+        return values
+
+    @root_validator(pre=True)
+    def valid_column_casting(cls, values):
```

Comment:
Common functionality to the tables one. 
Should move it into a util lib

Reply:
Same for the validations bellow.

Reply:
will do

Reply:
@eugenemiretsky got the subclassing working with SourceBaseConfig, so this is done